MantidQt/MantidWidgets/src/InstrumentView/ProjectionSurface.cpp

```diff
@@ -191,34 +197,20 @@ void ProjectionSurface::draw(MantidGLWidget *widget, bool picking) const {
 
     if (!picking) {
       QPainter painter(widget);
-      RectF windowRect = getSurfaceBounds();
-      m_maskShapes.setWindow(windowRect, painter.viewport());
-      m_maskShapes.draw(painter);
-      for (int i = 0; i < m_peakShapes.size(); ++i) {
-        m_peakShapes[i]->setWindow(windowRect, painter.viewport());
-        m_peakShapes[i]->draw(painter);
-      }
+      drawMaskShapes(painter);
```

Comment:
A nice bit of refactoring here. This looks much cleaner and and it's easier to follow what is happening.